### DISCUSSION TITLE
OS로 데비안 계열의 raspbian 을 사용하는게 수업 진행에 문제가 없을까요?

### DISCUSSION THREAD

**Discussion by kor-haru** [tedilabs/fastcampus-devops#70] [Q&A] [answered]
안녕하세요. 패캠 강의 수강자입니다.

남는 장비로 ARM 아키텍처인 라즈베리파이가 있고
기존에 docker 랑 docker-compose 까지 설치는 해둔 환경이 있어서
이 녀석으로 학습을 진행할까 하는데요
macOS 환경 구성에서도 M1 칩 기반 환경에 대한 언급은 없는거 같아서.... 
향후 k8s 실습엔 문제가 될까 싶어서 질문드립니다.

x86 기반 리눅스 환경을 권장하시는거로 이해하면 될까요??

**Comment by posquit0** [comment, marked as answer]
안녕하세요, 도커 컨테이너 기술은 리눅스를 기반으로 한 기술이라 macOS와 Windows 사용자의 경우 Docker Desktop 동작 방식이 다를 수 있어 강의 중 안내드렸습니다.
이야기해주신 ARM 아키텍처(M1 맥, 라즈베리파이 등)은 다른 CPU 아키텍처인 경우로 각각의 도구들이 ARM 컴파일 버전을 지원해주어야 할 듯 합니다.
docker, docker-compose, minikube 등 사용하시는 모든 도구들이 ARM을 지원하는지 여부와 실습에 사용하는 도커 이미지들이 ARM 기반으로 빌드된 것이 있는지도 중요할 것 같습니다.

가능한 이슈를 겪지 않고 실습을 진행하기 위해서는 x86 Linux 환경을 준비하시는 것을 추천드립니다.

**Reply by kor-haru** [reply]
답변 감사합니다 🙏
